lib/crystalball/execution_map.rb

```diff
@@ -30,7 +36,19 @@ def to_h
     # @param [Hash] initial list of cases
     def initialize(metadata: {}, cases: {})
       @cases = cases
-      @metadata = Metadata.new(type: self.class.name, **metadata)
+
+      version = metadata[:version].to_f
+
+      if cases.any?
+        version = 1.0 if version.zero?
+        guard_version_compatibility(version)
+      elsif version.nonzero?
+        guard_version_compatibility(version)
+      else
+        version = VERSION
+      end
```

Comment:
Map version heavily depends on configuration and particular use, so we shouldn't specify it. It should be specified in outer world instead via Configuration object same as `commit`.
E.g. 
```
MapGenerator.start! do |config|
  config.version = 1
end
```

P.S. Don't forget to check version consistency on map load

Reply:
It is checked now because https://github.com/toptal/crystalball/blob/master/lib/crystalball/map_storage/yaml_storage.rb#L39 checks the consistency of all metadata fields